airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingSqlOperations.java

```diff
@@ -54,44 +50,38 @@ public String getStagingPath(final String connectionId, final String namespace,
   }
 
   @Override
-  public void uploadRecordsToStage(final JdbcDatabase database, final File dataFile, final String schemaName, final String path) throws Exception {
-    throw new NotImplementedException("placeholder function is not implemented yet");
-  }
-
-  @Override
-  public void insertRecordsInternal(final JdbcDatabase database,
-                                    final List<AirbyteRecordMessage> records,
-                                    final String schemaName,
-                                    final String stage) {
-    LOGGER.info("Writing {} records to {}", records.size(), stage);
-
-    if (records.isEmpty()) {
-      return;
-    }
+  public String uploadRecordsToStage(final JdbcDatabase database,
+                                     final RecordBufferImplementation recordsData,
+                                     final String schema,
+                                     final String stage) {
     try {
-      loadDataIntoStage(database, stage, records);
+      loadDataIntoStage(database, stage, recordsData);
+      return recordsData.getFilename();
     } catch (final Exception e) {
       LOGGER.error("Failed to upload records into stage {}", stage, e);
       throw new RuntimeException(e);
     }
   }
 
-  private void loadDataIntoStage(final JdbcDatabase database, final String stage, final List<AirbyteRecordMessage> partition) throws Exception {
-    final File tempFile = Files.createTempFile(UUID.randomUUID().toString(), ".csv").toFile();
-    writeBatchToFile(tempFile, partition);
-    database.execute(String.format("PUT file://%s @%s PARALLEL = %d", tempFile.getAbsolutePath(), stage, Runtime.getRuntime().availableProcessors()));
-    Files.delete(tempFile.toPath());
+  private void loadDataIntoStage(final JdbcDatabase database, final String stage, final RecordBufferImplementation recordsData) throws Exception {
+    database.execute(
+        String.format("PUT file://%s @%s PARALLEL = %d", recordsData.getFile().getAbsolutePath(), stage, Runtime.getRuntime().availableProcessors()));
   }
 
   @Override
-  public void createStageIfNotExists(final JdbcDatabase database, final String stageName) throws SQLException {
+  public void createStageIfNotExists(final JdbcDatabase database, final String stageName) throws Exception {
+    final String query = "CREATE STAGE IF NOT EXISTS %s encryption = (type = 'SNOWFLAKE_SSE') copy_options = (on_error='skip_file');";
```

Comment:
this query is not used, but it's the same as https://github.com/airbytehq/airbyte/blob/eaae6049a4198f4e04ec36eba1bfa627dff18049/airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingSqlOperations.java#L21-L22 so probably fine to delete

Reply:
Oh yea i missed this when merging with another snowflake PR in master